### PR TITLE
Feat Disable auth methods and limit number of users.

### PR DESCRIPTION
**Pull request by eldadfux**
## What does this PR do?

Adds option to control and limit your Appwrite project auth methods.

- Disable email/password
- Disable anon login (need to merge and apply changed)
- Disable JWT
- Disable team invites
- Option to limit number of users (good for app launches + god account PR)
- Added 2 new endpoints to the projects API to allow new settings
- Added a new UI screen on the users section
- Enabled 501 errors (Not Implemented) from the error handler
- Added new e2e tests to validate new features

image

## Test Plan

Added new tests.

## Related PRs and Issues

#907 #385 #914 #210 #906 #472 #937 

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes.

### CODE REVIEW

**Review by TorstenDittmann — APPROVED** [review]
lgtm 👍🏻

File: app/controllers/shared/api.php [hunk 1]
```diff
@@ -110,6 +110,61 @@
...
+    /** @var Utopia\Registry\Registry $register */
+    /** @var Appwrite\Event\Event $events */
+    /** @var Appwrite\Event\Event $audits */
+    /** @var Appwrite\Event\Event $usage */
+    /** @var Appwrite\Event\Event $deletes */
+    /** @var Appwrite\Event\Event $functions */
+
+    $route = $utopia->match($request);
+
+    $isPreviliggedUser = Auth::isPreviliggedUser(Authorization::$roles);
```
Comment (kodumbeats) [inline app/controllers/shared/api.php:129]: Can we fix the spelling of this method in this PR or save for another?
https://github.com/appwrite/appwrite/blob/7acfa98221c1e2a6a96731aabd7254ee2b2ae6c0/src/Appwrite/Auth/Auth.php#L217
Should be isPrivilegedUser

File: app/controllers/general.php [hunk 2]
```diff
@@ -278,6 +278,7 @@
         case 409: // Error allowed publicly
         case 412: // Error allowed publicly
         case 429: // Error allowed publicly
+        case 501: // Error allowed publicly
```
Comment (kodumbeats) [inline app/controllers/general.php:281]: If we're using the standard 400 status codes, should we do the same for 500 (501 is Not Implemented)? It doesn't make a functional difference, so long as we respond with an appropriate error description ( like [here](https://github.com/appwrite/appwrite/pull/947/files#diff-93771e9bb8e57cbe7d0c0e91e7aea5d85238a362f51caf957049f1d2ccc4d312R91))

File: app/controllers/shared/api.php (hunk 1 above)
Comment (eldadfux) [reply app/controllers/shared/api.php:129]: I'll fix it in this one 👍🏻

File: app/controllers/general.php (hunk 2 above)
Comment (eldadfux) [reply app/controllers/general.php:281]: We allow 500 errors, but we hide the description with a generic message ("Server Error") to hide server errors in production.

**Review by kodumbeats — CHANGES_REQUESTED** [review]
All looks good once we correct `isPrivilegedUser` 📦 👍🏻

**Review by kodumbeats — APPROVED** [review]
<3

**Comment by TorstenDittmann** [comment]
Resolved conflicts and synced this branch 👍🏻

**Comment by boboB6** [comment]
Yes babe

**Pull request merged**